src/main/AppConnector.h

```diff
@@ -62,5 +62,9 @@ class AppConnector
     medida::MetricsRegistry& getMetrics() const;
     SearchableHotArchiveSnapshotConstPtr
     copySearchableHotArchiveBucketListSnapshot();
+
+    // Get a snapshot of ledger state for use by the overlay thread only. Must
+    // only be called from the overlay thread.
+    SearchableSnapshotConstPtr getOverlayThreadSnapshot();
```

Comment:
I'm not sure if we want to maintain a persistent, self updating BucketListSnapshot for the overlay thread, as self-updating snapshots have bitten us before. We should either just create a new snapshot every time, or if we do want to persist a snapshot for performance reasons, maybe we should manually make calls to update as necessary.

Reply:
That's fair. I assume that would mean calling `copySearchableLiveBucketListSnapshot` on every transaction though. Is that performant enough to call 1000+ times per second? This is an honest question; I can't tell whether `copySearchableLiveBucketrListSnapshot` makes a copy of the snapshot itself, or if it's making a copy of a pointer to a pre-existing immutable snapshot (fast), or something else.

Reply:
It's fairly cheap, it's just a pointer copy + opening up some new file stream. 1000+ times per seconds probably means the file streams will be expensive though. In that case I'd keep the single snapshot, but remove the auto-updating logic from the getter and just to it explicitly after getting the snapshot. That way if we have some logic in a future flow that requires a consistent snapshot between several loads, we won't run into this potential footgun.

Reply:
Got it. I kept the snapshot, but removed the auto-updating. Now I just update it before using it.